pkg/service/egresstracking/service.go

```diff
@@ -128,5 +144,149 @@ func (s *EgressTrackingService) egressTrack(ctx context.Context, batchCID cid.Ci
 		return fmt.Errorf("invocation failed: %s", x.Message)
 	}
 
+	// Extract the consolidate invocation from the receipt's effects
+	effects := rcpt.Fx()
+	if effects != nil {
+		fork := effects.Fork()
+		if len(fork) > 0 {
+			// The first effect should be the consolidate invocation
+			consolidateInvLink := fork[0].Link()
+
+			// Store the track invocation and consolidate CID (indexed by batch CID)
+			consolidateCID := consolidateInvLink.(cidlink.Link).Cid
+			if err := s.consolidationStore.Put(ctx, batchCID, trackInv, consolidateCID); err != nil {
+				return fmt.Errorf("storing track invocation in consolidation store: %w", err)
+			}
+			log.Infof("stored track invocation with consolidate invocation %s for batch %s", consolidateInvLink.String(), batchCID.String())
+		}
+	}
+
+	return nil
+}
+
+// StartCleanupTask starts the periodic cleanup task that checks for consolidated batches
+// and removes them from the store.
+func (s *EgressTrackingService) StartCleanupTask(ctx context.Context) error {
+	if s.cleanupCheckInterval <= 0 {
+		log.Info("cleanup task disabled (interval is 0)")
+		close(s.cleanupDone)
+		return nil
+	}
+
+	cleanupCtx, cancel := context.WithCancel(ctx)
+	s.cleanupCancel = cancel
+
+	go s.runCleanupTask(cleanupCtx)
+
+	log.Infof("cleanup task started with interval: %v", s.cleanupCheckInterval)
+	return nil
+}
+
+// StopCleanupTask stops the periodic cleanup task gracefully.
+func (s *EgressTrackingService) StopCleanupTask(ctx context.Context) error {
+	if s.cleanupCancel != nil {
+		s.cleanupCancel()
+	}
+
+	select {
+	case <-s.cleanupDone:
+		log.Info("cleanup task stopped")
+		return nil
+	case <-ctx.Done():
+		return fmt.Errorf("timeout waiting for cleanup task to stop: %w", ctx.Err())
+	}
+}
+
+func (s *EgressTrackingService) runCleanupTask(ctx context.Context) {
+	defer close(s.cleanupDone)
+
+	ticker := time.NewTicker(s.cleanupCheckInterval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("cleanup task context cancelled")
+			return
+		case <-ticker.C:
+			if err := s.cleanupConsolidatedBatches(ctx); err != nil {
+				log.Errorf("error cleaning up consolidated batches: %v", err)
+			}
+		}
+	}
+}
+
+func (s *EgressTrackingService) cleanupConsolidatedBatches(ctx context.Context) error {
+	// List all batches
+	batchCIDs, err := s.journal.List(ctx)
+	if err != nil {
+		return fmt.Errorf("listing batches: %w", err)
+	}
+
+	if len(batchCIDs) == 0 {
+		log.Debug("no batches to clean up")
+		return nil
+	}
+
+	log.Debugf("checking %d batches for consolidation", len(batchCIDs))
+
+	// Check each batch for consolidation
+	for _, batchCID := range batchCIDs {
+		if err := s.checkAndRemoveConsolidatedBatch(ctx, batchCID); err != nil {
+			log.Errorf("error checking batch %s: %v", batchCID, err)
+			// Continue with other batches even if one fails
+		}
+	}
```

Comment:
Noticing that we have an opportunity to perform this operation in parallel (https://pkg.go.dev/golang.org/x/sync/errgroup is handy for this sort of thing), but also get that we don't (know if we) need that yet. Considering adding a comment here for our future selves.